sdc/datatypes/hpat_pandas_dataframe_rolling_functions.py

```diff
@@ -78,43 +82,175 @@
 """
 
 
-def df_rolling_method_codegen(method_name, self, args=None, kws=None):
+def params2list(params):
+    """Converts parameters from dict to list"""
+    return ['{}={}'.format(k, v) for k, v in params.items()]
+
+
+def df_rolling_params_codegen():
+    """Generate rolling parameters"""
+    params = ['window', 'min_periods', 'center', 'win_type', 'on', 'axis', 'closed']
+    return ', '.join(f'self._{p}' for p in params)
+
+
+def df_rolling_method_with_other_df_codegen(method_name, self, other, args=None, kws=None):
     args = args or []
     kwargs = kws or {}
 
-    rolling_params = ['window', 'min_periods', 'center',
-                      'win_type', 'on', 'axis', 'closed']
-    rolling_params_as_str = ', '.join(f'self._{p}' for p in rolling_params)
-    method_params = args + ['{}={}'.format(k, k) for k in kwargs]
-    method_params_as_str = ', '.join(method_params)
-
-    impl_params = ['self'] + args + ['{}={}'.format(k, v) for k, v in kwargs.items()]
+    rolling_params = df_rolling_params_codegen()
+    method_kws = {k: k for k in kwargs}
+    impl_params = ['self'] + args + params2list(kwargs)
     impl_params_as_str = ', '.join(impl_params)
 
+    data_columns = {col: idx for idx, col in enumerate(self.data.columns)}
```

Comment:
Maybe this way?
```suggestion
    data_columns = dict(zip(self.data.columns, range(len(self.data.columns))))
```

Reply:
I don't see principle difference between these ways. @kozlov-alexey what is the purpose of the change in your opinion?

Reply:
I just thought that it looks better than dict comprehension, since there's no need to swap keys and values. I don't mind keeping it as-is though.